### ISSUE TITLE
Use Argon2d instead SHA2 as key derivation function 

### ISSUE THREAD

**Issue by LoHub** [cyb3rko/pincredible#26]
Hello,

First of all, thank you for developing PINcredible which look a great project.

I quickly read the code, in particular the import/export function.
As far as I can understand, the password typed by the user is hashed by SHA2 function and the hash is used as AES key to encrypt serialized data. Unfortunately modern GPUs can do billion SHA2 hash per second.

In my opinion, a good improvement could be to use Argon2d function instead of SHA2, which will make dictionary attacks harder on exported file. Argon2d is build to be hard to crack even if user password have poor entropy and it is very efficient against GPUs usage.

Links :

* https://crypto.stackexchange.com/questions/50135/is-sha-2-suitable-for-key-derivation#answer-50137
* https://en.wikipedia.org/wiki/Argon2
* https://github.com/lambdapioneer/argon2kt (example of implementation in Kotlin)

**Comment by cyb3rko** [comment]
@LoHub Argon2ID is now implemented, it will be included in the next release.  
I've used the following specifications:  
- mode: Argon2id
- 65.536 KibiByte memory usage
- 10 iterations
- salt of size 32 Bytes
- parallelism of 2 threads

---

The only thing that's weird is that the key derivation is blazing fast compared to the implementation of the [lambdapioneer/argon2kt](https://github.com/lambdapioneer/argon2kt) author's demo app.  
Where his app needs 2,5 seconds mine is done in less than 400 milliseconds.
